sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp

```diff
@@ -35,41 +39,77 @@ namespace Azure { namespace Core { namespace Http {
     constexpr static size_t DefaultUploadChunkSize = 1024 * 64;
     constexpr static size_t MaximumUploadChunkSize = 1024 * 1024;
 
-    struct HandleManager final
+    // unique_ptr class wrapping an HINTERNET handle
+    class HINTERNET_deleter {
```

Comment:
These aren't our types. These are types defined by Windows and we should not rename existing types (That's the same reason we don't define String as an alias for std::string, even though std::string doesn't follow our naming guidelines). Similarly, we don't rename other platform types to conform to our naming guidelines.

Reply:
We don't capitalize Url, Http, Guid, Sha. We don't have any examples of defining a type with the name like you're proposing:
It has
1. More than 2 capital letters => we change it to Hinternet_deleter
2. We use PascalCase (deleter starts with lowercase) => we change it to Hinternet_Deleter
3. We don't use underscores and snake_case => we change it to HinternetDeleter.

Reply:
Guid, Sha, and Url aren't types defined by other components, they are concepts.

`CURL`, `HINTERNET`, `CERTCONTEXT`, etc are types defined by libcurl, WInHTTP, and BCrypt respectively.

Each of these types is a specialization of `std::unique_ptr` on the specified type. I need a named specialization because I wanted to make it clear that these types are std::unique_ptr (and thus have all the semantics of `std::unique_ptr`.

Reply:
https://azure.github.io/azure-sdk/cpp_introduction.html#naming